code/game/machinery/firealarm.dm

```diff
@@ -273,35 +278,10 @@
 		new /obj/item/stack/cable_coil(loc, 3)
 	qdel(src)
 
-
-/*
- * Party button
- */
-
-/obj/machinery/firealarm/partyalarm
-	name = "\improper PARTY BUTTON"
-	desc = "Cuban Pete is in the house!"
-
-
-/obj/machinery/firealarm/partyalarm/reset()
-	if (stat & (NOPOWER|BROKEN))
-		return
-	var/area/A = src.loc
-	A = A.loc
-	if (!( istype(A, /area) ))
-		return
-	A.partyreset()
-
-/obj/machinery/firealarm/partyalarm/alarm()
-	if (stat & (NOPOWER|BROKEN))
-		return
-	var/area/A = src.loc
-	A = A.loc
-	if (!( istype(A, /area) ))
-		return
-	A.partyalert()
-
-/obj/machinery/firealarm/partyalarm/ui_data(mob/user)
-	. = ..()
-	var/area/A = get_area(src)
-	.["alarm"] = A.party
+/obj/machinery/firealarm/proc/update_fire_light(fire)
+	if(fire == !!light_power)
+		return  // do nothing if we're already active

```

Comment:
This is confusing to read, should probably just be dropped since set_light() already has code for this iirc

Reply:
It still gets queued, only the heavy part of the update is skipped when SSlighting eventually gets to it.